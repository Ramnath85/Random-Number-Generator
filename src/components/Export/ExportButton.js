import React from 'react';
import PropTypes from 'prop-types';

const ExportButton = ({ phoneNumbers, onClick }) => (
  <div className="input-wrapper">
    {
      phoneNumbers.length > 0 &&
      <button
        className="export-button"
        onClick={onClick}>
        Export Numbers
      </button>
    }
  </div>
);

ExportButton.propTypes = {
  phoneNumbers: PropTypes.array,
  onClick: PropTypes.func,
};

export default ExportButton;