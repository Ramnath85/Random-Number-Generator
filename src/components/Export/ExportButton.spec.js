import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ExportButton from "./ExportButton";

configure({ adapter: new Adapter() });

describe('Test ExportButton component', () => {
  let shallowComponent;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      phoneNumbers: [],
      onClick: () => {}
    };
    shallowComponent = shallow(<ExportButton {...defaultProps} />);
  });

  test('should render the Export button', () => {
    expect(shallowComponent).toMatchSnapshot();
    expect(shallowComponent.length).toEqual(1);
  });

  test('should render export text', () => {
    shallowComponent.setProps({
      phoneNumbers: ['0903428765', '0787665456']
    });
    expect(shallowComponent.find('button').text()).toContain('Export')
  });
});