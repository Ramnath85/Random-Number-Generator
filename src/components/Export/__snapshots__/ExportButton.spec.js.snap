// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ExportButton component should render the Export button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExportButton
    onClick={[Function]}
    phoneNumbers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "input-wrapper",
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "input-wrapper",
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
